engine/src/flutter/lib/web_ui/lib/src/engine/semantics/scrollable.dart

```diff
@@ -183,64 +172,51 @@ class SemanticScrollable extends SemanticRole {
     }
   }
 
-  /// Resets the scroll position (top or left) to the neutral value.
-  ///
-  /// The scroll position of the scrollable HTML node that's considered to
-  /// have zero offset relative to Flutter's notion of scroll position is
-  /// referred to as "neutral scroll position".
-  ///
-  /// We always set the scroll position to a non-zero value in order to
-  /// be able to scroll in the negative direction. When scrollTop/scrollLeft is
-  /// zero the browser will refuse to scroll back even when there is more
-  /// content available.
-  void _neutralizeDomScrollPosition() {
+  /// Updates the scrollable state of the semantics object.
+  void _updateScrollableState() {
     // This value is arbitrary.
-    const int canonicalNeutralScrollPosition = 10;
     final ui.Rect? rect = semanticsObject.rect;
     if (rect == null) {
       printWarning('Warning! the rect attribute of semanticsObject is null');
       return;
     }
     if (semanticsObject.isVerticalScrollContainer) {
-      // Place the _scrollOverflowElement at the end of the content and
-      // make sure that when we neutralize the scrolling position,
-      // it doesn't scroll into the visible area.
-      final int verticalOffset = rect.height.ceil() + canonicalNeutralScrollPosition;
+      // Place the _scrollOverflowElement at the beginning of the content
+      // and size it based on the total scroll extent so the browser
+      // knows how much scrollable content there is.
+      final double? scrollExtentMax = semanticsObject.scrollExtentMax;
+      final double? scrollExtentMin = semanticsObject.scrollExtentMin;
+      assert(scrollExtentMax != null);
+      assert(scrollExtentMin != null);
+      final double scrollExtentTotal = scrollExtentMax! - scrollExtentMin! + rect.height.round();
```

Comment:
Should we `round()` after adding all the doubles? I.e. `(a + b + c).round()` instead of `a + b + c.round()`

Reply:
Maybe we shouldn't round here at all since we want to match the frameworks calculation of total scroll extent https://github.com/flutter/flutter/blob/110b07835ab17e6aea29c6d192649b6fa48e4092/packages/flutter/lib/src/widgets/scroll_metrics.dart#L154. What do you think?

Reply:
Are we trying to account for a precision error?

Reply:
I don't know enough to say if precision matters here. However, since this is stringified into `'${scrollExtentTotal}px'`, I'd run it through `toStringAsFixed`, e.g. `'${scrollExtentTotal.toStringAsFixed(1)}px'`.